proposals/0216-widget-support.md

```diff
@@ -246,73 +263,268 @@ The RPC `Show` and `SetDisplayLayout` are identified as the only screen related
+
+Details about the currently selected screen template are part of the display capabilities struct:
+- text fields, the name, width etc.
+- image fields e.g. graphic, or soft button image resolutions
+
+The template design comes from the head unit. However predefined template names are defined in `PredefinedLayout` enum. The head unit should provide template designs for the predefined template names. Still, it can also provide additional custom templates with self defined names.
+
+All this can be reused for widget screens. When creating a new widget screen, the available widget templates are provided in `ScreenCapability.screenLayoutAvailable`.
+
+The details about the currently selected widget template are provided using the same display capabilities struct. It is expected that the HMI will return with fewer text fields (only main field 1 and 2), shorter text field width and smaller image resolutions. This solution reuses the existing API, to provide widget template details as mentioned.
+
+Examples for widget screens:
+
+| template name | Example layout |
+|-|-|
+| text with graphic  | ![template](../assets/proposals/NNNN-widgets/template-text-with-graphic.png) |
+| graphic with text  | ![template](../assets/proposals/NNNN-widgets/template-graphic-with-text.png) |
+| buttons with graphic | ![template](../assets/proposals/NNNN-widgets/template-tiles-with-graphic.png) |
 
 ### Policies
 
-With above modification of `OnHMIStatus` the existing policies are compatible to widgets. The policy manager should be modified and use the HMI level of the main screen to evaluate permissions for all RPCs, except screen specific RPCs. In this case the HMI level of the targeting screen should be used for evaluation.
+With above modification of `OnHMIStatus` the existing policies are compatible to widgets. The policy manager of Core should be modified and use the HMI level of the main screen to evaluate permissions for all RPCs, except screen specific RPCs. In this case the HMI level of the targeting screen should be used for evaluation.
+
+A new functional group should be added that reflects permissions of the new RPCs.
+
+```json
+"functional_groupings": {
+  …
+  "WidgetSupport" : {
+      "rpcs":{
+          "CreateScreen":{
+              "hmi_levels":["NONE","BACKGROUND","LIMITED","FULL"]
+          }, 
+          "DeleteScreen":{
+              "hmi_levels":["NONE","BACKGROUND","LIMITED","FULL"]
+          }
+      }
+  }
+}
+```
+
+This functional group should be provided to the app developers as a conditional feature, they can request. This should be done on the SmartDeviceLink developer portal when managing the app and the permissions. The web portal needs to have a checkbox additionally to the other checkboxes. The checkbox could be labeled "Allow creating widgets".
```

Comment:
Update to:
This functional group should be provided to the app developers as a conditional feature, they can request. **This will need to be added on the SmartDeviceLink Developer Portal when creating a "New App". The "New App" page needs to have a checkbox under "Special Permissions Required for Integration" labeled "Allow Creating Widgets".**